### ISSUE TITLE
False positive: for kv in Dict(x)

### ISSUE THREAD

**Issue by TotalVerb** [tonyhffong/Lint.jl#181]
```julia
julia> lintstr("""
       for kv in Dict(1=>2)
           @show kv
       end
       """)
1-element Array{Lint.LintMessage,1}:
 none:1 I474 Tuple{Int64,Int64}: iteration generates tuples, 1 of 2 variables used
```

I don't think we should raise the I474 here. If the user is not expecting a key-value pair, then there is probably some other type error in the code that we can catch instead.
